x-pack/plugins/security_solution/public/network/store/reducer.ts

```diff
@@ -190,4 +200,66 @@ export const networkReducer = reducerWithInitialState(initialNetworkState)
       queries: setNetworkDetailsQueriesActivePageToZero(state),
     },
   }))
+  .case(updateNetworkAnomaliesJobIdFilter, (state, { jobIds, networkType }) => {
+    if (networkType === NetworkType.page) {
+      return {
+        ...state,
+        page: {
+          ...state.page,
+          queries: {
+            ...state.page.queries,
+            anomalies: {
+              ...state.page.queries.anomalies,
+              jobIdSelection: jobIds,
+            },
+          },
+        },
+      };
```

Comment:
I believe we could use `set` from _lodash/fp_ (not from the normal lodash) instead in all these big reducers:
```suggestion
      return set('page.queries.anomalies.jobIdSelection', jobIds, state);
```
Since the _fp_ lib implements immutability, it won't mutate the current state and return a new one.